packages/it-parallel-batch/index.js

```diff
@@ -2,26 +2,43 @@
 
 const batch = require('it-batch')
 
+/**
+ * @template T
+ * @typedef {function():Promise<T>} Task
+ */
+
+/**
+ * Takes an (async) iterator that emits promise-returning functions,
+ * invokes them in parallel and emits the results as they become available but
+ * in the same order as the input
+ *
+ * @template T
+ * @param {AsyncIterable<Task<T>>} source
+ * @param {number|string} [size=1]
+ * @returns {AsyncIterable<T>}
+ */
 async function * parallelBatch (source, size) {
+  // @ts-ignore - expects string not a number
   size = parseInt(size)
 
   if (isNaN(size) || size < 1) {
     size = 1
   }
 
-  for await (let things of batch(source, size)) {
-    things = things.map(p => {
-      return p().then(res => ({ res }), err => ({ err }))
+  for await (const tasks of batch(source, size)) {
+    /** @type {Promise<{ok:true, value:T}|{ok:false, err:Error}>[]} */
+    const things = tasks.map(p => {
+      return p().then(value => ({ ok: true, value }), err => ({ ok: false, err }))
```

Comment:
📝 Type checker can refine union types as long as they share a common property unique per variant. That is why `ok` field was added. That way type checker can infer type of `result` in both branches of `if (result.ok)`.